include/irr/asset/IAssetManager.h

```diff
@@ -434,11 +434,15 @@ class IAssetManager : public core::IReferenceCounted
         /** Keeping assets around (by their pointers) helps a lot by letting the loaders retrieve them from the cache 
 		and not load cpu objects which have been loaded, converted to gpu resources and then would have been disposed of.
 		However each dummy object needs to have a GPU object associated with it in yet-another-cache for use when we convert CPU objects to GPU objects.*/
-        void convertAssetToEmptyCacheHandle(IAsset* _asset, core::smart_refctd_ptr<core::IReferenceCounted>&& _gpuObject, uint32_t referenceLevelsBelowToConvert=~0u)
+        void convertAssetToEmptyCacheHandle(IAsset* _asset, core::smart_refctd_ptr<core::IReferenceCounted>&& _gpuObject, uint32_t referenceLevelsBelowToConvert=0u)
         {
+            if (_asset->isDummyObjectForCacheAliasing)
+                return;
+
 			const uint32_t ix = IAsset::typeFlagToIndex(_asset->getAssetType());
             _asset->grab();
             _asset->convertToDummyObject(referenceLevelsBelowToConvert);
+            _asset->isDummyObjectForCacheAliasing = true;
```

Comment:
doesn't ` _asset->convertToDummyObject` do that `= true` thing already?

Reply:
some of them, actually it seems a little like we decided to put `=true` in IAsset::convertToDummyObject implementations but then forgot about it and now some of them have this and some dont

Reply:
that should definitely be inside the implementations!

Reply:
related https://github.com/buildaworldnet/IrrlichtBAW/pull/410#discussion_r368248021

Reply:
fixed in today commit (not commited/pushed yet while writing this)